components/payments/cmd/connectors/internal/connectors/modulr/task_fetch_transactions.go

```diff
@@ -81,7 +167,7 @@ func fetchTransactions(
 		}
 	}
 
-	return nil
+	return state, nil
```

Comment:
You should return newState here, when testing, the new state is never inserted in the database